src/SerilogWeb.Classic/Classic/ApplicationLifecycleModule.cs

```diff
@@ -130,34 +133,59 @@ public static LogEventLevel FormDataLoggingLevel
         /// <summary>
         /// Initializes a module and prepares it to handle requests.
         /// </summary>
-        /// <param name="context">An <see cref="T:System.Web.HttpApplication"/> that provides access to the methods, properties, and events common to all application objects within an ASP.NET application </param>
-        public void Init(HttpApplication context)
-        {
-            context.LogRequest += LogRequest;
-            context.Error += Error;
-        }
-
-        static void LogRequest(object sender, EventArgs e)
+        /// <param name="application">An <see cref="T:System.Web.HttpApplication"/> that provides access to the methods, properties, and events common to all application objects within an ASP.NET application </param>
+        public void Init(HttpApplication application)
         {
-            if (!_isEnabled) return;
-
-            var request = HttpContextCurrent.Request;
-            if (request == null)
-                return;
+            application.BeginRequest += (sender, args) =>
+            {
+                if(_isEnabled && application.Context != null)
+                {
+                    application.Context.Items[StopWatchKey] = Stopwatch.StartNew();
+                }                
+            };
 
-            Logger.Write(_requestLoggingLevel, "HTTP {Method} for {RawUrl}", request.HttpMethod, request.RawUrl);
-            if (ShouldLogRequest())
+            application.EndRequest += (sender, args) =>
```

Comment:
`HttpContext.Items` is available in `LogRequest`, I chose the events that I did based on the referenced code. Based on my review of the ASP.net lifecycle, either `EndRequest` or `LogRequest` are both feasible choices to stop the stopwatch and log the timing, and I don't think either provide additional "accuracy" for this use case.

It is also feasible to log the error inside of `EndRequest`, I just kept it in the same event hook as the original module, only changing the syntax so all of the methods matched.